pkg/volume/csi/csi_plugin.go

```diff
@@ -892,34 +891,33 @@ func highestSupportedVersion(versions []string) (*utilversion.Version, error) {
 		return nil, errors.New(log("CSI driver reporting empty array for supported versions"))
 	}
 
-	// Sort by lowest to highest version
-	sort.Slice(versions, func(i, j int) bool {
-		parsedVersionI, err := utilversion.ParseGeneric(versions[i])
-		if err != nil {
-			// Push bad values to the bottom
-			return true
-		}
-
-		parsedVersionJ, err := utilversion.ParseGeneric(versions[j])
-		if err != nil {
-			// Push bad values to the bottom
-			return false
-		}
-
-		return parsedVersionI.LessThan(parsedVersionJ)
-	})
-
+	var highestSupportedVersion *utilversion.Version
+	seenNoErr := false
+	var theErr error
 	for i := len(versions) - 1; i >= 0; i-- {
-		highestSupportedVersion, err := utilversion.ParseGeneric(versions[i])
+		currentHighestVer, err := utilversion.ParseGeneric(versions[i])
 		if err != nil {
-			return nil, err
+			theErr = err
+			continue
+		} else {
+			seenNoErr = true
 		}
-
-		if highestSupportedVersion.Major() <= 1 {
-			return highestSupportedVersion, nil
+		if currentHighestVer.Major() > 1 {
+			continue
+		}
+		if highestSupportedVersion == nil {
+			highestSupportedVersion = currentHighestVer
+		} else if highestSupportedVersion.LessThan(currentHighestVer) {
+			highestSupportedVersion = currentHighestVer
 		}
 	}
 
+	if highestSupportedVersion != nil {
+		return highestSupportedVersion, nil
+	}
+	if theErr != nil && !seenNoErr {
```

Comment:
Previously this code didn't return the parsing error. Do we really want to change that? Can we instead expand the generic error below to include 1) the list of versions returned by the driver and 2) any errors in parsing?